pointcloud/basic_octree.go

```diff
@@ -26,9 +32,10 @@ type NodeType uint8
 // and serialization.
 type BasicOctree struct {
 	node       basicOctreeNode
-	center     r3.Vector
+	center     spatialmath.Pose
```

Comment:
I dont think this should be a Pose as orientation has nothing to do with an octree's implementation

Reply:
I disagree here. How is it that something with a location unable to have an orientation? Especially when that thing is full of three dimensional points.

Reply:
The other option which might be nicer, would be to remove `center` entirely, have an octree implicitly centered at its own (0,0,0), and create a wrapper which is able to place it at some `pose` and wrap the transformation math there.

What we have currently is a limited, specific implementation that draws a circle around a subset of spatial transformations and says "these are the ones that we're going to make easy to do". The notion of an orientation has exactly as much to do with a specific octree implementation as the notion of an octree having a position.